### ISSUE TITLE
Citus upgrade fails from 10.0.3 to 10.1.2

### ISSUE THREAD

**Issue by hlakshmi** [citusdata/citus#5473]
The upgrade path from 10.0.3 to 10.1.2 is broken due to the following error. This happens only in a specific scenario where a PG upgrade is also involved. Following are the sequence of steps:

1. We are Postgres version 11.4 and Citus version 10.0.3
2. Upgraded Postgres to 13.4
3. Now try upgrading Citus to 10.1.2 ----> This FAILS

Following is the error when running the command `ALTER EXTENSION citus UPDATE`:
```
#<PG::UndefinedObject: ERROR:  constraint "chunk_storage_id_stripe_num_chunk_group_num_fkey" of relation "chunk" does not exist
CONTEXT:  SQL statement "
ALTER TABLE columnar.chunk DROP CONSTRAINT chunk_storage_id_stripe_num_chunk_group_num_fkey;
ALTER TABLE columnar.chunk_group DROP CONSTRAINT chunk_group_storage_id_stripe_num_fkey;
  "
PL/pgSQL function inline_code_block line 4 at EXECUTE
```
This happens because of the change here: https://github.com/citusdata/citus/blob/master/src/backend/columnar/sql/columnar--10.0-3--10.1-1.sql

**Root Cause**
The foreign key creation on the above columnar tables happens when the Postgres version is 11.4 and hence they have the PG11 convention.  Now in step (3) when we are trying to upgrade Citus to 10.1.2, the server version is 13.4 and the migration SQL tries to follow drop these constraints following PG13 convention and results in the above failure.

**Comment by onurctirtir** [comment]
Let me try to reproduce the issue and get back to you today.

**Comment by onurctirtir** [comment]
Btw, thank you for sharing the root cause. Even without trying to reproduce the issue, your the root cause explanation realized me that we (actually I :) ) couldn't think of the scenario, where pg version is upgraded and our migration script tries to drop the foreign key based on new postgres version.

Doing SQL changes in older versions is mostly not something we prefer, so I'm not sure if/when we can fix that, but let me share the following workaround to unblock you:

Could you try renaming those foreign keys so that their names will be converted to what script expects for that postgres version, before running `ALTER EXTENSION citus UPDATE` ?

Or, could you first upgrade citus, then the postgres version ?

**Comment by hlakshmi** [comment]
Thanks @onurctirtir for the response. Yes, that's one of the monkey-patch which we have thought about but wanted to avoid messing up with the tables under Citus schema so the later upgrades can go smoothly.

**Comment by onurctirtir** [comment]
Yeah you're right a lot. Then let me first reproduce the issue, and test the workaround that I suggested.
I will let you know once I finish doing that.

**Comment by onurctirtir** [comment]
Verified that running following commands on all nodes before doing `ALTER EXTENSION citus UPDATE` on those nodes mitigates the issue:

```sql
ALTER TABLE columnar.chunk RENAME CONSTRAINT chunk_storage_id_fkey TO chunk_storage_id_stripe_num_chunk_group_num_fkey;
ALTER TABLE columnar.chunk_group RENAME CONSTRAINT chunk_group_storage_id_fkey TO chunk_group_storage_id_stripe_num_fkey;
```

**Comment by onurctirtir** [comment]
Hi @hlakshmi, did the workaround help you ?

**Comment by hlakshmi** [comment]
Thanks @onurctirtir,  I am going to try it out today and get back to you.

**Comment by hlakshmi** [comment]
@onurctirtir FYI I have tried the workaround which you provided and it worked fine.

**Comment by onurctirtir** [comment]
Sounds great!

**Comment by hlakshmi** [comment]
@onurctirtir We are eagerly looking for a permanent fix for this as we can't modify/patch the Citus code base due to licensing reason (which is AGPL). Can you let us know if there's a plan ahead to fix this issue for the affected versions?
Thanks!
